src/generator/protocol/operations.ts

```diff
@@ -353,6 +353,29 @@ function createProtocolRequest(client: string, op: Operation, imports: ImportMan
     if (bodyParam!.schema.type === SchemaType.Constant) {
       // if the value is constant, embed it directly
       body = formatConstantValue(<ConstantSchema>bodyParam!.schema);
+    } else if (mediaType === 'XML' && bodyParam!.schema.serialization?.xml) {
+      // for XML payloads, create a wrapper type in the following cases
+      // 1. the type's name doesn't match the XML name (e.g. "Slideshow" vs. "slideshow")
+      // 2. the payload is a wrapped array
+      if (bodyParam!.schema.language.go!.name !== bodyParam!.schema.serialization.xml.name || bodyParam!.schema.type === SchemaType.Array) {
+        imports.add('encoding/xml');
+        text += '\ttype wrapper struct {\n';
+        text += `\t\tXMLName xml.Name \`xml:"${bodyParam!.schema.serialization.xml.name}"\`\n`;
+        let fieldName = bodyParam!.schema.language.go!.name;
+        if (isArraySchema(bodyParam!.schema)) {
+          fieldName = pascalCase(bodyParam!.language.go!.name);
+          let tag = bodyParam!.schema.elementType.language.go!.name;
+          if (bodyParam!.schema.elementType.serialization?.xml?.name) {
+            tag = bodyParam!.schema.elementType.serialization.xml.name;
+          }
+          text += `\t\t${fieldName} *${bodyParam!.schema.language.go!.name} \`xml:"${tag}"\`\n`;
+        } else {
+          // embed as anonymous field
+          text += `\t\t*${bodyParam!.schema.language.go!.name}\n`;
```

Comment:
Just trying to understand this a little more, what cases do we embed this anonymously?

Reply:
If it's an object then it's embedded anonymously (there are only two cases, arrays or objects).

Reply:
Ah ok, good to know. Thanks!